### PR TITLE
feat: rework add-choice UX (explained types, Multi-as-folder, add-into-folder) + fix AI-disabled layout

### LINKED ISSUE #1256: Rework add-choice UX (clearer types, Multi-as-folder) + fix bottom-bar layout when AI disabled
## Bug
In the settings Choices view, when **AI & Online features are disabled**, the add-choice bar collapses hard to the left. Root cause: `.choiceViewBottomBar` uses `justify-content: space-between`, so the lone add box (no AI Assistant button to balance against) snaps left.

## Rework
The add flow (`[Name] [Template/Capture/Macro/Multi ▾] [Add Choice]`) presents the four types as an unexplained dropdown, and "Multi" — really a **folder** — masquerades as a peer type. This is confusing for newcomers.

Planned (Direction C + grafts):
- A top action bar: `+ New choice ▾` (Obsidian Menu of the 3 *doer* types, each with a one-line description) + a separate **`+ New folder`** button. Multi becomes structurally not-a-peer; the layout bug becomes impossible.
- Rename Multi → "folder" in user-facing copy (stored value stays `"Multi"`, no migration).
- Auto-open the relevant ChoiceBuilder after adding a doer (Alt-click to skip).
- First-class **add-into-folder**: per-folder `+` affordance + context-menu entry.
- Empty/first-run state.

Shipped in phases (one PR): P1 layout decouple, P2 explained add controls, P3 add-into-folder + empty state.

### PR DESCRIPTION
**Pull request by chhoumann**
Closes #1256.

Reworks how users **add a choice** in the settings Choices view, and fixes the reported layout bug. Built on top of #1255 (Obsidian 1.13 declarative settings).

## The problem
The old add row — `[Name] [Template/Capture/Macro/Multi ▾] [Add Choice]` — presented the four types as an unexplained dropdown, and **"Multi" (really a folder) masqueraded as a peer type**. And when **AI/online features were disabled**, the add box collapsed hard to the left (`.choiceViewBottomBar` used `justify-content: space-between`, so the lone add box had nothing to balance against).

## What changed
- **Top action bar:** `+ New choice ▾` — an Obsidian menu of the three *doer* types, each with a one-line description — plus a **separate `+ New folder`** button. Multi is no longer a peer "type"; it reads as a folder. Renamed Multi → "folder" in user-facing copy (**stored `ChoiceType` value unchanged — no migration**).
- **First-class add-into-folder:** a per-folder add affordance inside each expanded folder, rendered **outside the dndzone** (so svelte-dnd never treats it as a draggable item / shadow placeholder). Adds route through the **top-level** handler via the exported `insertIntoMulti`/`addChoiceToTree`, so nested adds persist correctly at any depth (the `rootReorder` invariant).
- **Builder handoff:** after adding a doer, its builder opens automatically (**Alt-click skips** for batch scaffolding); folders skip it. Auto-named, so the flow never gates on typing a name first.
- **Empty/first-run state** with a single hero CTA (no duplicate controls).
- **Layout fix:** the add controls are now a left-anchored, content-width cluster (`margin-right: auto`) with the AI button pinned right — the add cluster's position is **independent of the AI button**, so it can't collapse when AI is disabled.
- **a11y:** WCAG 2.5.3 label-in-name on the add buttons; `aria-haspopup`/`aria-expanded` on the menu trigger.
- Deletes `AddChoiceBox.svelte`; `MultiChoice` now initializes `collapsed = false`.

## Verification
**CI-equivalent locally:** `tsc`, `svelte-check` (0 errors), `lint`, **1470 tests** — including new `addChoiceToTree`/`insertIntoMulti` unit tests and a **ChoiceView add-into-folder integration test** that drives the per-folder affordance through the real DOM (asserts the new choice lands nested and the folder stays expanded).

**Real Obsidian 1.13.0 dev vault** (eval + `data.json` assertions): top bar renders `flex`; add cluster left at content-width while the **AI button pins far right** when on and the cluster doesn't move when off (bug fixed in both states); `New folder` and per-folder add persist (nested at depth 1); zero runtime errors.

**Adversarial review** (4 agents) ran on the diff and verified the load-bearing invariants hold (dndzone isolation, top-level `insertIntoMulti` persistence with no depth≥2 corruption, snapshot plainness, no `./main` value-import). Its recommended items are all addressed in this PR: the add-into-folder test, the empty-state double-render, WCAG labels, and dropping the eager double-save.

## Scope notes
- **Deferred** the context-menu "New choice here" entry — the per-folder affordance already covers add-into-folder and is keyboard-accessible.
- The `+ New choice` **menu → pick-type → builder** flow and the **empty state** can't be *synthetically* driven (Obsidian menus don't open via programmatic clicks — verified the existing context menu behaves identically), so those are covered by unit tests + parity rather than a scripted end-to-end.

---

<a href="https://app.devin.ai/review/chhoumann/quickadd/pull/1257" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * New add-controls: “New choice” menu and “New folder” action, Alt/⌥ to skip configuration, autogenerated default names, responsive placement, and improved empty-state CTA; supports adding into folders.

* **Refactor**
  * Consolidated and relocated add UI into unified controls; add-row exposed for multi-folder lists.

* **Bug Fixes & Stability**
  * More reliable add/save flow, auto-expands target folders and scrolls new items into view.

* **Tests**
  * Expanded unit, lifecycle and accessibility tests covering add controls and tree insertion.

[Review Change Stack](https://app.coderabbit.ai/change-stack/chhoumann/quickadd/pull/1257?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/gui/choiceList/ChoiceView.svelte [hunk 1]
```diff
@@ -100,10 +101,37 @@
...
+		// isn't stranded on an unconfigured row; handleConfigureChoice persists
+		// the result, so there's no eager save here (avoids a double write).
+		// Folders and Alt-click skip the builder and are committed immediately.
+		if (type !== "Multi" && !skipConfigure) {
+			try {
+				await handleConfigureChoice(newChoice);
+			} catch (err) {
+				// Builders resolve rather than reject, but don't let a stray throw
+				// become an unhandled rejection or lose the new choice.
+				console.error("QuickAdd: failed to configure new choice", err);
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/gui/choiceList/ChoiceView.svelte]: Fixed in 105c65f — switched to `log.logError` (per AGENTS.md's "log through the logger utilities" convention). Thanks!

File: src/gui/choiceList/ChoiceView.svelte [hunk 2]
```diff
@@ -100,10 +101,37 @@
...
+				save();
+			}
+		} else {
+			save();
+		}
 	}
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/gui/choiceList/ChoiceView.svelte:160]: Addressed the eager-save half: doers now persist via `handleConfigureChoice` only (no double write), and the catch routes through `log.logError`. The deeper point — the builders resolve `waitForClose` with the choice rather than rejecting on cancel, so a cancelled add still persists a default-configured choice — is a pre-existing builder limitation (no cancel signal), tracked as a follow-up rather than expanded in this PR.

File: src/gui/choiceList/ChoiceView.svelte [hunk 3]
```diff
@@ -210,66 +261,118 @@
...
+	}
+
+	/* The add cluster is pinned RIGHT (margin-left:auto); the AI button (when
+	   shown) sits to its left. The add cluster's position is independent of
+	   whether the AI button is present, so it can't collapse when AI/online
+	   features are disabled (the original justify-content:space-between bug). */
+	.choiceAddGroup {
+		display: flex;
+		gap: 0.5rem;
+		margin-left: auto;
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/gui/choiceList/ChoiceView.svelte]: Deliberate design call: after comparing left- vs right-anchored layouts in the dev vault (a guided design pass), the maintainer chose the right-packed bottom bar. The empty-gutter concern is handled by `justify-content: flex-end` (no `margin-left:auto`) plus demoting the AI button to a quiet icon, so the bar neither floats nor shifts when AI toggles. Keeping it right-anchored.
